lib/r10k/git/rjgit.rb

```diff
@@ -60,6 +60,41 @@ def rev_parse(ver, opts = {})
         return commit.id
       end
 
+      def fetch(remote, opts={})
+
+        local = opts[:cachedir]
+        repo = ::RJGit::Repo.new(local, is_bare: true)
+
+        opts[:refspecs] = "+refs/*:refs/*"
+        begin
+          repo.git.fetch(remote, opts)
+        rescue Java::OrgEclipseJgitApiErrors::GitAPIException,
+               Java::OrgEclipseJgitApiErrors::TransportException,
+               Java::OrgEclipseJgitApiErrors::InvalidRemoteException => e
+          raise R10K::Git::GitError.new(e.message)
+        end
+
+        return true
+      end
+
+      def clone(remote, opts={})
+        opts[:branch] = :all
```

Comment:
this is going to alter the state of opts which seems pretty bad given how they're being fed around.